asmcli/commands/validate.sh

```diff
@@ -37,17 +38,23 @@ validate_dependencies() {
 
   if can_register_cluster && is_gcp; then
     register_cluster
-  elif should_validate && [[ "${USE_HUB_WIP}" -eq 1 ]]; then
+  elif should_validate && [[ "${USE_HUB_WIP}" -eq 1 || "${USE_VM}" -eq 1 ]]; then
     exit_if_cluster_unregistered
   fi
 
   if is_gcp; then
     if can_modify_gcp_components; then
       enable_workload_identity
       enable_stackdriver_kubernetes
+      if should_enable_service_mesh_feature; then
+        enable_service_mesh_feature
+      fi
     else
       exit_if_no_workload_identity
       exit_if_stackdriver_not_enabled
+      if should_enable_service_mesh_feature; then
```

Comment:
Let's not use `should_` here. We need a new function `needs_servicemesh_feature` like `needs_asm`. The fact that we require something is separate from whether we should enable it or not.

Reply:
Done